src/observable/map/ApplyMap.ts

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2020 Bruno Windels <bruno@windels.cloud>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import {BaseObservableMap, BaseObservableMapConfig} from "./BaseObservableMap";
+import {SubscriptionHandle} from "../BaseObservable";
+import {config, Mapper, Updater, Comparator, Filter} from "./config";
+import {JoinedMap} from "./JoinedMap";
+import {MappedMap} from "./MappedMap";
+import {FilteredMap} from "./FilteredMap";
+import {SortedMapList} from "../list/SortedMapList.js";
+
+
+export class ApplyMap<K, V> extends BaseObservableMap<K, V> {
+    private _source: BaseObservableMap<K, V>;
+    private _subscription?: SubscriptionHandle;
+    private _apply?: Apply<K, V>;
+    private _config: BaseObservableMapConfig<K, V>;
```

Comment:
All classes which `extend` `BaseObservableMap` get this private config variable.